L2/hardfork_devnet_test.md

```diff
@@ -0,0 +1,94 @@
+# Devnet Hard Fork Testing Guide: Enabling L2Blob and Soul Gas Token (SGT)
+
+This document provides instructions for performing hard fork upgrades on QuarkChain's OP Stack devnet, specifically to enable L2Blob and Soul Gas Token (SGT) functionalities post-initial deployment.
+
+## 1. Launch the devnet with L2Blob and SGT disabled
+Deploy the OP Stack according to the instructions provided [here](https://github.com/QuarkChain/pm/blob/main/L2/opup_devnet_test.md).
+
+### Notes:
+  - When editing `intent.toml`, ensure you remove the following fields to disable the respective features during initial deployment.
+    - `soulGasTokenTimeOffset` 
+    - `l2GenesisBlobTimeOffset`
+### Validation after deployment:
+  - **SGT Validation**: Executing "Spend SGT without native gas token" should result in:
+    ```bash
+    Error: Failed to estimate gas: server returned an error response: error code -32000: gas required exceeds allowance (0)
+    ```
+  - **L2Blob Validation**: Transactions involving L2Blob should yield:
+    ```bash
+    Error: server returned an error response: error code -32000: transaction type not supported
+    ```
+## 2. Updating the Superchain Registry
+Perform the following steps to update the superchain registry:
+  - Clone and checkout the devnet branch
+    ```bash
+    git clone https://github.com/QuarkChain/superchain-registry.git
+    cd superchain-registry
+    git checkout devnet
+    ```
+  - Update configuration files:
+      - Modify `./chainList.toml` and `./chainList.json` based on your deployment specifics.
```

Comment:
Looks like these two files are automatically generated and not actually used for op-geth/op-node.